server/src/main/java/org/elasticsearch/index/codec/tsdb/ES87TSDBDocValuesProducer.java

```diff
@@ -903,12 +903,17 @@ private static DocValuesSkipperEntry readDocValueSkipperMeta(IndexInput meta) th
         return new DocValuesSkipperEntry(offset, length, minValue, maxValue, docCount, maxDocID);
     }
 
-    private static void readNumeric(IndexInput meta, NumericEntry entry) throws IOException {
-        entry.docsWithFieldOffset = meta.readLong();
-        entry.docsWithFieldLength = meta.readLong();
-        entry.jumpTableEntryCount = meta.readShort();
-        entry.denseRankPower = meta.readByte();
+    private static void readNumeric(IndexInput meta, NumericEntry entry, int version) throws IOException {
+        if (version < ES87TSDBDocValuesFormat.VERSION_TWO) {
```

Comment:
Since these branches are far away, should we fork a new consumer and producer instead?

Reply:
done: 378ee29f436736422d116ae14ae6074d6c5bd826